libs/SalesforceSDK/src/com/salesforce/androidsdk/config/BootConfig.java

```diff
@@ -109,74 +130,92 @@ public static void validateBootConfig(BootConfig config) {
 			throw new BootConfigException("No boot config provided.");
 		}
 
-		if (SalesforceSDKManager.getInstance().isHybrid()) {
-			String startPage = config.getStartPage();
-			if (startPage == null) {
-				throw new BootConfigException("No start page configured.");
+		if (config.configIsHybrid) {
+			// startPage must be a relative URL.
+			if (BootConfig.isAbsoluteUrl(config.getStartPage())) {
+				throw new BootConfigException("Start page should not be absolute URL.");
 			}
 
-			// Sanity check local URLs against absolute URL values.
-			if (config.isLocal() && config.isStartPageAbsoluteUrl()) {
-				throw new BootConfigException("Local start page should not be absolute URL.");
+			// unauthenticatedStartPage doesn't make sense in a local setup.  Warn accordingly.
+			if (config.isLocal() && config.getUnauthenticatedStartPage() != null) {
+				SalesforceSDKLogger.w(TAG, UNAUTHENTICATED_START_PAGE + " set for local app, but it will never be used.");
 			}
 
-			// Start page makeup for remote apps beyond this point is subject to conditional configuration.
+			// unauthenticatedStartPage doesn't make sense in a remote setup with authentication.  Warn accordingly.
+			if (!config.isLocal() && config.shouldAuthenticate() && config.getUnauthenticatedStartPage() != null) {
+				SalesforceSDKLogger.w(TAG, UNAUTHENTICATED_START_PAGE + " set for remote app with authentication, but it will never be used.");
+			}
 
-			if (!config.isLocal() && config.shouldAuthenticate() && config.isStartPageAbsoluteUrl()) {
-				throw new BootConfigException("Cannot authenticate using an absolute URL for start page.");
+			// Lack of unauthenticatedStartPage with remote deferred authentication is an error.
+			if (!config.isLocal() && !config.shouldAuthenticate()
+					&& (config.getUnauthenticatedStartPage() == null || config.getUnauthenticatedStartPage().length() == 0)) {
+				throw new BootConfigException(UNAUTHENTICATED_START_PAGE + " required for remote app with deferred authentication.");
 			}
 
-			if (!config.isLocal() && !config.shouldAuthenticate() && !config.isStartPageAbsoluteUrl()) {
-				throw new BootConfigException("Cannot configure relative URL start page with deferred authentication.");
+			// unauthenticatedStartPage, if present, must be an absolute URL.
+			if (config.getUnauthenticatedStartPage() != null
```

Comment:
Minor: You could use `!TextUtils.isEmpty(config.getUnauthenticatedStartPage())` here instead if you like - it checks for `null` or empty string.